ansible/ansible-ai-connect-service
# retry SSO token service

Jira Issue: https://issues.redhat.com/browse/AAP-18946

## Description
AMS health checks are failing in production due to the SSO token service calls timing out. This PR adds a retry and bumps up the timeout on this call. It also makes the timeout and number of retries configurable.

## Testing

### Steps to test
1. Pull down the PR
2. Set AUTHZ_SSO_TOKEN_SERVICE_TIMEOUT = "0.2"
3. Hit the /check/status endpoint
The 0.2 timeout seems like the right threshold to sometimes pass and sometimes fail, and you can see the backoff working successfully. Check /metrics to confirm authz_sso_token_service_retries_total is present and non-zero.

### Scenarios tested
Steps described above, plus tested with a 401 by using wrong client id and confirming the error flow worked correctly.

## Production deployment
